### ISSUE TITLE
Request: option for fixing "Working: No" without destroying logs

### ISSUE THREAD

**Issue by Taomyn** [huginn/huginn#2843]
Could we get an option to fix the need to destroy all the logs of an agent just because there was a single "failed" at any point in the logs, even if later the agent is working fine?

Either a reset of the "No" to "Yes" or the ability to delete individual log entries, or something else.

**Comment by dsander** [comment]
Normally the working state should flip back to yes after the Agent processed or produced an Event successfully. Which is the Agent you are having trouble with?

**Comment by Taomyn** [comment]
Haven't been tracking which types of agent, although most of mine are a mix of RSS and Website agents.

One this morning is an RSS agent, I re-ran it and it's still the same:
![image](https://user-images.githubusercontent.com/4365929/87758988-d57b9e80-c80d-11ea-9da7-41cd0ce62a96.png)

![image](https://user-images.githubusercontent.com/4365929/87759103-03f97980-c80e-11ea-9c95-ea87add6dee7.png)

I have not cleared the logs for this one yet which is what I would normally do

**Comment by dsander** [comment]
Understood, the Agent is shown as not working because an error happened after it emitted it's last Event. I agree that it kind of makes sense that the error that happened on the third last run should not matter because it worked two times after the error happened. The problem I see with a change like that is that it would hide a lot of failures.

**Comment by Taomyn** [comment]
I agree it shouldn't clear the issue automatically, just create a way to reset it manually otherwise if I even fix the root cause I cannot clear the error without destroying all the history.

**Comment by dsander** [comment]
IC, I don't think there is an easy way to do this because Agents have different rules when it's considered "working". What could be an option is to allow the `last_error_log_at`, `last_check_at` etc attributes.

**Comment by Taomyn** [comment]
Actually on further reflection in light of other system issues I have had recently (none to so with Huginn) I think it would be better if "Working" simply reflected the status of the last time it was executed - transient issues should not affect this. I can't think of any cases on my various servers where the current status is set to an error state because in the distant past it failed. Sure, log and report it when it does fail, but then when it works again the state should be good.

**Comment by dsander** [comment]
Many Agents work like this, some require an Event to be emitted after the last error was reported (which can or can not make sense depending on the specific job the Agent has). Take a `WebsiteAgent` for example, it failed to check the site, then it ran again but didn't emit an Event. It runs a few more times without emitting an Event. The Agent might be doing it's job correctly because nothing changed, or doesn't because the structure of the site changed so that the Agent will never emit an Event again. 

That's why the Agents `working?` condition is `event_created_within(options['expected_update_period_in_days']) && !recent_error_logs?`, i.e. it emitted something and didn't create an error after emitting the last Event.

I think most Agents working conditions are based on the idea that a `working?` Agent is most probably doing its job while something _might_ be off for a not working Agent.
